pipeline/lib/metric.ml

```diff
@@ -0,0 +1,134 @@
+module Github = Current_github
+module BenchmarksData = Map.Make (String)
+
+let parse_benchmark_data ~data =
+  Array.fold_left
+    (fun acc datum ->
+      match datum with
+      | [| _commit; benchmark_name; test_name; metrics |] ->
+          let key = Fmt.str "%s__%s" benchmark_name test_name in
+          let parsed_metrics = Yojson.Safe.from_string metrics in
+          BenchmarksData.add key parsed_metrics acc
+      | _ -> acc)
+    BenchmarksData.empty data
+
+let sum_float_list xs = List.fold_left (fun acc x -> acc +. x) 0. xs
+let calc_diff v cv = Float.abs (v -. cv) /. cv *. 100.
```

Comment:
Perhaps this function should be responsible for checking `cv <> 0.0` ? (otherwise this responsibility is pushed to the caller, and if we ever change the logic of `calc_diff` such that this precondition is not required anymore... For example I think we might have to do `Float.abs cv > 0.0001` rather than `cv <> 0.0` because division by near zero can quickly yield huge numbers even though we don't really care about this tiny metric)

Reply:
Makes sense. Fixed this! :+1: